src/host.c

```diff
@@ -0,0 +1,519 @@
+/**
+ * collectd - src/host.c
+ * Copyright (C) 2019       Graham Leggett
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ * Authors:
+ *   Graham Leggett <minfrin at sharp.fm>
+ *
+ **/
+
+/*
+ * This plugin keeps track of hosts that have been seen by collectd, and sends
+ * notifications when a new host is seen, and when a host has not been seen.
+ *
+ */
+
+#include "collectd.h"
+
+#include "plugin.h"
+#include "utils/common/common.h"
+
+#include <lmdb.h>
+
+#define PLUGIN_NAME "host"
+
+#define DEFAULT_HOST_PATH "hosts"
+#define DEFAULT_LOST_INTERVAL 10
+#define DEFAULT_THREAD_INTERVAL 2
+#define DEFAULT_DELAY_INTERVAL 10
+
+static const char *config_keys[] = {"HOSTPATH", "LOSTINTERVAL",
+                                    "DELAYINTERVAL"};
+static int config_keys_num = STATIC_ARRAY_SIZE(config_keys);
+static char *host_path = DEFAULT_HOST_PATH;
+static int lost_interval = DEFAULT_LOST_INTERVAL;
+static int delay_interval = DEFAULT_DELAY_INTERVAL;
+static int thread_interval = DEFAULT_THREAD_INTERVAL;
+
+static pthread_mutex_t host_lock = PTHREAD_MUTEX_INITIALIZER;
+static pthread_cond_t host_cond = PTHREAD_COND_INITIALIZER;
+static int host_thread_loop;
+static pthread_t host_thread_id;
+
+static MDB_env *env;
+
+static void *host_thread(void *arg) /* {{{ */
+{
+  struct timeval tv_begin = {0};
+
+  if (gettimeofday(&tv_begin, NULL) < 0) {
+    ERROR("host plugin: gettimeofday failed: %s", STRERRNO);
```

Comment:
Here you added prefix to messages, later there are many cases w/o prefix.

Reply:
I have made them consistent in dcd1bb5, can you verify for me?